Iterable-iOS-SDK/IterableAPI.m

```diff
@@ -683,6 +683,23 @@ - (void)updateUser:(NSDictionary *)dataFields mergeNestedObjects:(BOOL)mergeNest
     }
 }
 
+// documented in IterableAPI.h
+- (void)updateEmail:(NSString *)newEmail onSuccess:(OnSuccessHandler)onSuccess onFailure:(OnFailureHandler)onFailure
+{
+    if (newEmail) {
+        NSDictionary *args = @{
+                ITBL_KEY_CURRENT_EMAIL: self.email,
+                ITBL_KEY_NEW_EMAIL: newEmail
+        };
+
+        NSURLRequest *request = [self createRequestForAction:ENDPOINT_UPDATE_EMAIL withArgs:args];
+        [self sendRequest:request onSuccess:^(NSDictionary *dictionary) {
+            _email = newEmail;
+            onSuccess(dictionary);
```

Comment:
i think you need to wrap this in a `if (onSuccess != nil) onSuccess(dictionary)`; if you look at the definition of `sendRequest` it does that